### PR TITLE
Add Super pipeline for code transforms.

### LINKED ISSUE #173: [Feature] Implement a super-pipeline for Code processing
### Search before asking

- [X] I searched the [issues](https://github.com/IBM/data-prep-lab/issues) and found no similar issues.

### Component

KFP workflows

### Feature

Implement a super pipeline that will combine all code-processing steps.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by revit13**
/Closes #173 

## Why are these changes needed?

This PR implements a pipeline for the [notebook ](https://github.com/IBM/data-prep-kit/blob/dev/examples/demo_with_launcher.ipynb)

ingest 2 parquet phase is not part of the super pipeline. its output was manually generated by running the notebook phase.

Please note that transforms images are not synced with the latest code in the repo. To run the pipelines please do:
 `cd transforms && make image && load-image` to upload the latest versions to kind cluster.

## Related issue number (if any).
https://github.com/IBM/data-prep-kit/issues/173

### CODE REVIEW

**Comment by roytman** [comment]
@revit13 , we'll need a document that explains it

**Comment by revit13** [comment]
> @revit13 , we'll need a document that explains it

@roytman I added description to kfp/doc/multi_transform_pipeline.md.

File: kfp/doc/multi_transform_pipeline.md [hunk 1]
```diff
@@ -23,13 +23,30 @@ In the list of its input parameters, we also see `data_s3_config`. Now, we have
 
 ![param list](param_list2.png)
 
+### Examples
 
-**Note** An example super pipeline that combines several transforms, `doc_id`, `ededup`, and `fdedup`, can be found in [superworkflow_dedups_sample_wf.py](../superworkflows/v1/superworkflow_dedups_sample_wf.py).
+The sections that follow display two super pipelines as examples:
+
+1) [dedups super pipeline](#De-dups-super-pipeline)
```
Comment (roytman) [inline kfp/doc/multi_transform_pipeline.md]: I think the link will not work.
better to use explicit names.
```
### Dedups super pipeline <a name = "dedups"></a>
```
so the link can be [dedups super pipeline](#dedups)

File: kfp/doc/multi_transform_pipeline.md [hunk 2]
```diff
@@ -23,13 +23,30 @@ In the list of its input parameters, we also see `data_s3_config`. Now, we have
...
+
+This pipeline combines several transforms, `doc_id`, `ededup`, and `fdedup`, can be found in [superworkflow_dedups_sample_wf.py](../superworkflows/ray/kfp_v1/superworkflow_dedups_sample_wf.py).
 
 ![super pipeline](super_pipeline.png)
 
+The input parameters of the super pipelines are described in this [section](#super-pipeline-Input-Parameters).
+
+### Programming languages Super pipeline
+
+This pipeline combines several programming-languages transforms: `ededup`, `doc_id`, `fdedup`, `proglang_select`, `code_quality`,  `malware` and `tokenization`. It can be found in [superworkflow_code_wf.py](../superworkflows/ray/kfp_v1/superworkflow_code_sample_wf.py).
```
Comment (roytman) [inline kfp/doc/multi_transform_pipeline.md]: This pipeline combines transforms for programming languages data preprocessing:

File: kfp/superworkflows/ray/kfp_v1/superworkflow_code_sample_wf.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+malware_image = "quay.io/dataprep1/data-prep-kit/malware:0.4.0"
+doc_id_image = "quay.io/dataprep1/data-prep-kit/doc_id:0.3.1"
+ededup_image = "quay.io/dataprep1/data-prep-kit/ededup:0.3.0"
+fdedup_image = "quay.io/dataprep1/data-prep-kit/fdedup:0.3.0"
+tokenizer_image = "quay.io/dataprep1/data-prep-kit/tokenization:0.3.0"
+
+
+# Pipeline to invoke execution on remote resource
+@dsl.pipeline(
+    name="sample-super-kubeflow-pipeline",
```
Comment (roytman) [inline kfp/superworkflows/ray/kfp_v1/superworkflow_code_sample_wf.py]: maybe to change the pipeline name and description to the specific usecase.

File: kfp/doc/multi_transform_pipeline.md (hunk 1 above)
Comment (revit13) [reply kfp/doc/multi_transform_pipeline.md]: Fixed. Thanks

File: kfp/doc/multi_transform_pipeline.md (hunk 2 above)
Comment (revit13) [reply kfp/doc/multi_transform_pipeline.md]: Fixed. Thanks

File: kfp/superworkflows/ray/kfp_v1/superworkflow_code_sample_wf.py (hunk 3 above)
Comment (revit13) [reply kfp/superworkflows/ray/kfp_v1/superworkflow_code_sample_wf.py]: Fixed. Thanks

**Review by roytman — APPROVED** [review]
LGTM

**Pull request merged**